addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java

```diff
@@ -0,0 +1,154 @@
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.security.RetentionExpiredFinderListener;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.DefaultRepositoryInit;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.retention.adapters.RetentionRule;
+import org.nuxeo.retention.adapters.RetentionRule.StartingPointPolicy;
+import org.nuxeo.retention.service.RetentionManager;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ TransactionalFeature.class, AutomationFeature.class })
+@RepositoryConfig(init = DefaultRepositoryInit.class, cleanup = Granularity.METHOD)
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-core-types.xml")
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-adapters.xml")
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-vocabularies.xml")
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-content-template.xml")
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-service-framework.xml")
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-listeners.xml")
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-operations.xml")
+public abstract class RetentionTestCase {
+
+    @Inject
+    protected RetentionManager service;
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected BulkService bulkService;
+
+    protected DocumentModel file;
+
+    @Before
+    public void setup() {
+        file = session.createDocumentModel("/", "File", "File");
+        file = session.createDocument(file);
+        file = session.saveDocument(file);
+    }
+
+    protected void assertStillUnderRetentionAfter(DocumentModel doc, RetentionRule rule, int timeoutMillis)
+            throws InterruptedException {
+        doc = service.attachRule(doc, rule, session);
+        assertTrue(doc.isRecord());
+
+        awaitRetentionExpiration(timeoutMillis);
+
+        doc = session.getDocument(doc.getRef());
+
+        // it is still under retention and has a retention date
+        assertTrue(session.isUnderRetentionOrLegalHold(doc.getRef()));
+        assertNotNull(session.getRetainUntil(doc.getRef()));
+    }
+
+    protected void awaitRetentionExpiration(long millis) throws InterruptedException {
+        // wait a bit more than retention period to pass retention expiration date
+        coreFeature.waitForAsyncCompletion();
```

Comment:
I am wondering if we can use `transactionalFeature.nextTransaction();` instead of `coreFeature.waitForAsyncCompletion();` my idea is mainly about the consistency as mainly I used `transactionalFeature.nextTransaction()` when I wait for asynchronous things.